src/cli/services/linter/utils/deduplicateResults.ts

```diff
@@ -0,0 +1,26 @@
+import { IRange } from '@stoplight/types/dist';
+import { IRuleResult } from '../../../../types';
+
+type Dictionary<T, K extends PropertyKey> = { [key in K]: T };
+
+const ARTIFICIAL_ROOT = Symbol('root');
+
+const serializeRange = ({ start, end }: IRange) => `${start.line}:${start.character}:${end.line}:${end.character}`;
+const getIdentifier = (result: IRuleResult) => `${result.path.join('/')}${result.code}${serializeRange(result.range)}`;
```

Comment:
@P0lip Just an idea. How about turning that into a comparisonFunction (cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort)?

It could be used to detect duplicates in that use case (when `=== 0`) and reused in `stylish` to order the results.

Reply:
My sincere apologies for dropping a ball on this one.
Your new WIP PR reminded me of this PR.
Seems like you've started some work on that, so how do you feel about merging this one and implementing the comparison fn in your PR?

Reply:
>Seems like you've started some work on that

Well... I was merely scratching an itch to ensure my proposal was actually bringing _some_ kind of value. :wink: 

>so how do you feel about merging this one and implementing the comparison fn in your PR?

@P0lip :+1: